documentation/src/test/java/snippets/UniNullTest.java

```diff
@@ -21,6 +20,22 @@ public void uni() {
         assertThat(uni.onItem().ifNull().continueWith("hello").await().indefinitely()).isEqualTo("hello");
     }
 
+    @Test
+    public void uniNotNull() {
+        Uni<String> uni = Uni.createFrom().item(() -> null);
+        // tag::code-not-null[]
+        uni
+                .onItem().ifNotNull().apply(String::toUpperCase)
+                .onItem().ifNull().continueWith("yolo!");
+        // end::code-not-null[]
+
+        String r = uni
+                .onItem().ifNotNull().apply(String::toUpperCase)
+                .onItem().ifNull().continueWith("yolo!")
+                .await().indefinitely();
+        assertThat(r).isEqualTo("yolo!");
```

Comment:
I must be not grokking as I don't get how this would pass.

`uni` has `null` and therefore will return `yolo!` on subscription.

So I expected that `r` would return `YOLO!` and not `yolo!`.

Reply:
The uni produces `null`, so `ifNotNull().apply(String::toUpperCase)` is not called, and `null` is propagated downstream. As a consequence `.onItem().ifNull().continueWith("yolo!")` is called, `yolo!` is emitted. 

Does that make sense?

Reply:
But if you switch the 2 lines, it will become `YOLO!`, right?

(Because it's a pipeline, even though it really doesn't look like one.)

Reply:
Exactly.

Reply:
Ah, so are you saying `onItem().ifNull()` in the `uni` is never called because the previous line already consumed the `onItem()`?

If so, how would that line ever get executed?

Reply:
Each line is a stage, so there is an item emitted on every line. 

```
Uni.createFrom().item(() -> null); // emits `null`
  .onItem().ifNotNull().apply(String::toUpperCase) // skipped because the item is null, so emits (another) `null`
  .onItem().ifNull().continueWith("yolo!") // called because the received item is `null`, and emits "yolol!"
```

Reply:
Maybe my confusion is that the code is actually duplicated? In that the same `onItem()` calls are added to the same `uni` which means it's a single chain of 4 possible outcomes, as opposed to 2 chains with 2 possible outcomes each.

Am I reading that right now?

Reply:
So, yes the code is duplicated, as the first one (does nothing - no one subscribes) is just there for the snippet inclusion in the doc.

The second one is actually checking that it does the right thing.

Reply:
If I'm understanding, you could remove lines 33 and 34 and just do `uni.await()` right, as the `onItem()` entries are still in the Uni?

Reply:
No, every line returns a new Uni. So if line 27 we affect a new variable, and reuse it, then yes.